src/main/java/seedu/address/model/transaction/Budget.java

```diff
@@ -57,11 +82,46 @@ private void updateDeadline(Date date) {
     }
 
     /**
-     * Calculates the new Date given the amount of duration from Today
+     * Calculates the new Date given the amount of duration from Today.
      * @return Date after {@code duration} days from today
      */
     private Date calculateDeadline(int duration) {
-        return new Date("10102019");
+        LocalDate today = LocalDate.now();
+        LocalDate newDeadline = today.plus(duration, DAYS);
+        return new Date(newDeadline.toString());
+    }
+
+    /**
+     * Checks if the given budget is the same Budget object as this budget.
+     * @param otherBudget
+     * @return
+     */
+    public boolean isSameBudget(Budget otherBudget) {
+        if (otherBudget == this) {
+            return true;
+        }
+
+        return otherBudget != null
```

Comment:
@yewon0303 How about the tags? And to be more consistent with equals methods.

`return otherBudget != null`
`    && amount.equals(otherBudget.amount)`
    `&& deadline.equals(otherBudget.deadline)`
    `&& tags.equals(otherBudget.tags);`